errors/error.go

```diff
@@ -490,17 +506,46 @@ func (e Error) Log(fields log.Fields) {
 	fields.Add("err", e.errorCode.Desc())
 	fields.Add("errorCode", e.errorCode.Code())
 
-	if e.cause != nil {
-		fields.Add("cause", e.Error())
+	stackErr, ok := e.cause.(stackTracer)
+	if ok {
+		fields.Add("stack", stackErr.StackTrace())
+	}
+
+	causerErr, ok := e.cause.(causer)
+	if ok {
+		cause := causerErr.Cause()
+		loggableCause, ok := cause.(Error) // May need to split loggable interface.
+		if ok {
+			loggableCause.Log(fields)
+		}
+		fields.Add("cause", cause)
+	} else {
+		if e.cause != nil {
+			fields.Add("cause", e.cause)
+		}
 	}
 }
 
-// Cause implementation offset Err
+// Cause implementation of Error
 func (e Error) Cause() error {
+	causerErr, ok := e.cause.(causer)
+	if ok {
```

Comment:
```suggestion
	if causerErr, ok := e.cause.(causer); ok {
```
feel free to ignore if you don't like, but this is just my stylistic preference since it keeps the two defined variables (`causerErr`, `ok`) scoped to just the `if` block. Not that important in these specific cases where the methods are short but useful as a general convention especially when you have longer methods.

 Suggestion applies to `StackTrace` below as well.

Reply:
I like this, thanks! updated accordingly